src/main/scala/lc/Main.scala

```diff
@@ -70,26 +69,21 @@ class Captcha {
   }
 
   val task = new Runnable {
-  	val providerMap = getProvider()
-  	val provider = filters(providerMap)
   	def run(): Unit = {
-	    val challenge = provider.returnChallenge()
-	    val blob = new ByteArrayInputStream(challenge.content)
-	    val token = scala.util.Random.nextInt(10000).toString
-	    val id = Id(token)
-	    insertPstmt.setString(1, token)
-	    insertPstmt.setString(2, provider.getId)
-	    insertPstmt.setString(3, challenge.secret)
-	    insertPstmt.setString(4, providerMap)
-	    insertPstmt.setString(5, challenge.contentType)
-	    insertPstmt.setBlob(6, blob)
-	    insertPstmt.executeUpdate()
+      val imageNum = stmt.executeQuery("SELECT COUNT(*) AS total FROM challenge")
+      var throttleIn = throttle + ((10*throttle)/100).asInstanceOf[Int]
+      if(imageNum.next())
+        throttleIn = (throttleIn-imageNum.getInt("total"))
+      while(0 <= throttleIn-1){
```

Comment:
Could be simplified to `1 <= throttleIn`

Reply:
Yeah, it originally was `0 < throttleIn`. This is a minor debugging detail i missed.